proxy/src/control/discovery.rs

```diff
@@ -55,8 +63,17 @@ pub struct DiscoveryWork<T: HttpService<ResponseBody = RecvBody>> {
     rx: mpsc::UnboundedReceiver<(FullyQualifiedAuthority, mpsc::UnboundedSender<Update>)>,
 }
 
+#[derive(Clone, Debug, Hash, Eq, PartialEq)]
+pub struct DstLabels(Arc<str>);
+
+#[derive(Clone, Debug, Eq, PartialEq, Hash)]
+pub struct Labeled<T> {
+    metric_labels: Option<DstLabels>,
```

Comment:
As we discussed last night, this is generic over `T` because it's used both to annotate a discovery entry with a label _and_ as the middleware service that adds the label extension to requests. I can document that.

I agree with your comment regarding additional metadata --- maybe the discovery entry type should be changed from `Labeled<SocketAddr>` to
```rust
struct DiscoveryEntry { // i'm not particularly attached to this name
    addr: SocketAddr,
    metric_labels: Option<DstLabels>,
}
```
or something along those lines? 

Then the `Labeled` type can be used only as the service that's responsible for labeling requests...